### PR TITLE
Actually fix bad debug logging rejecting device list & signing key transactions

### PR DESCRIPTION
**Pull request by DMRobertson**
Missed in #11890.

### CODE REVIEW

File: tests/federation/transport/test_server.py [hunk 1]
```diff
@@ -38,3 +38,19 @@ def test_open_public_room_list_over_federation(self):
             "/_matrix/federation/v1/publicRooms",
         )
         self.assertEqual(200, channel.code)
+
+    @DEBUG
+    def test_edu_debugging_doesnt_explode(self):
+        """Sanity check fed. RX succeeds with `synapse.debug_8631` logging enabled.
```
Comment (babolivier) [inline tests/federation/transport/test_server.py:44]: I don't know what "RX" means, could you update this comment please?

File: changelog.d/12098.bugfix [hunk 2]
```diff
@@ -0,0 +1 @@
+Fix a bug introduced in Synapse 1.51.0rc1 where incoming federation transactions containing at least one EDU would be dropped if debug logging was enabled for `synapse.8631_debug`. Synapse 1.53.0rc1 included a partial fix this ([#11890](https://github.com/matrix-org/synapse/pull/11890)), but the fix was incorrect: EDUs containing device list or signing key updates would still be dropped.
```
Comment (babolivier) [inline changelog.d/12098.bugfix:1]: Not sure it's worth mentioning #11890 since it didn't end up fixing it?

**Review by babolivier — APPROVED** [review]
lgtm otherwise, modulo CI

File: tests/federation/transport/test_server.py (hunk 1 above)
Comment (DMRobertson) [reply tests/federation/transport/test_server.py:44]: TX: transmit
RX: receive

not sure where the abbreviation comes from. Synapse uses the former in `transaction_manager.py` and `per_destination_queue.py`. Slightly surprised to see it doesn't use the latter.

Comment (DMRobertson) [reply tests/federation/transport/test_server.py:44]: Possible etymology: https://en.wikipedia.org/wiki/RS-232

File: changelog.d/12098.bugfix (hunk 2 above)
Comment (DMRobertson) [reply changelog.d/12098.bugfix:1]: I just want to clarify that I've not accidentally copied and pasted a bugfix line from 1.53's release notes.

Comment (babolivier) [reply changelog.d/12098.bugfix]: Fair enough, though I'd tend to think looking at the PR is enough to tell that

File: tests/federation/transport/test_server.py (hunk 1 above)
Comment (babolivier) [reply tests/federation/transport/test_server.py]: Thanks! fwiw I just read `TX` in those file as an abbreviated version of `txn` for "transaction", TIL TX and RX mean transmit and receive.

**Review by babolivier — APPROVED** [review]
Sytest seems to be failing

**Comment by DMRobertson** [comment]
> Sytest seems to be failing

Yes, though in the same way that develop is failing. #12100 is tackling this.

**Pull request merged**